lightning/src/util/test_utils.rs

```diff
@@ -238,35 +237,39 @@ pub struct TestPersister {
 impl TestPersister {
 	pub fn new() -> Self {
 		Self {
-			update_ret: Mutex::new(chain::ChannelMonitorUpdateStatus::Completed),
-			next_update_ret: Mutex::new(None),
+			update_rets: Mutex::new(VecDeque::new()),
 			chain_sync_monitor_persistences: Mutex::new(HashMap::new()),
 			offchain_monitor_updates: Mutex::new(HashMap::new()),
 		}
 	}
 
+	/// Clear the queue of update statuses and set the one we'll always return.
 	pub fn set_update_ret(&self, ret: chain::ChannelMonitorUpdateStatus) {
-		*self.update_ret.lock().unwrap() = ret;
+		let mut update_rets = self.update_rets.lock().unwrap();
+		update_rets.clear();
+		update_rets.push_front(ret);
 	}
 
-	pub fn set_next_update_ret(&self, next_ret: Option<chain::ChannelMonitorUpdateStatus>) {
-		*self.next_update_ret.lock().unwrap() = next_ret;
+	/// Queue an update status to return.
+	pub fn set_next_update_ret(&self, next_ret: chain::ChannelMonitorUpdateStatus) {
+		self.update_rets.lock().unwrap().push_back(next_ret);
 	}
```

Comment:
Is the change in semantics here intentional? Previously, only calling `set_next_update_ret` would result in first returning `Completed` and then repeatedly returning `next_ret`. Now, it results in repeatedly returning `next_ret` only.

Reply:
Not really intentional, the intention was to make this util support the tests added in #1916 while still supporting existing tests